### PR TITLE
Wrap format validator to support nullable ("x-nullable") properties

### PR DESCRIPTION
**Pull request by adamhadani**
Currently only the type validator is wrapped to support the x-nullable workaround for null values, but the jsonschema format validator still fails on fields defined as such. This adds a wrapper around format validator that behaves very similarly to the type one

### CODE REVIEW

**Comment by adamhadani** [comment]
@asottile your review appreciated!

**Comment by asottile** [comment]
I uh, I'm going to be honest I don't really maintain this project much.  Most of my contributions are drive-by.  But @sjaensch or @macisamuele might be better choices.

One thing they're probably going to ask for is a test or two demonstrating your change :)

**Comment by adamhadani** [comment]
got it @asottile  thanks for input! added unit-test module, ping @sjaensch / @macisamuele any feedback appreciated!

**Comment by macisamuele** [comment]
Hey @adamhadani  sorry for the really long delay. 
Could you give me a concrete use-case in which you need it?
The code change it seems really straight forward and I would have a better idea of the potential impact.
Thanks a lot

**Comment by adamhadani** [comment]
@macisamuele sure let me try - basically we have a schema that defines a field of type 'string' and format 'uuid'. However, it should be nullable. the type validator has allowance for that, but it seems like the format validator did not, so I just added code similar to the one that was existing in the type validator to explicitly make allowance for null values.

**Comment by macisamuele** [comment]
Now I see what you mean ... thanks a lot for the use-case/explanation!
Before merging the request I would ask you to add an additional test like [this](https://github.com/macisamuele/bravado-core/blob/686857b96be4e678afc3c92683fb1410a17c5866/tests/swagger20_validator/enum_validator_test.py#L64).
Doing so I'll be sure that the change that you are proposing is actually fixing the issue and we'll ensure that future code changes will not break the feature.

Additionally will be great if you can fill also the PR information in our CHANGELOG as described on [Contribution](https://github.com/Yelp/bravado-core#contributing) section of the readme.

Thanks a lot for your contribution.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -2,6 +2,7 @@ Changelog
 =========
 X.X.X (XXXX-XX-XX)
 ------------------
+- Added support for nullable fields in the format validator
```
Comment (adamhadani) [inline CHANGELOG.rst]: ping @macisamuele added this comment here, is that what you meant?

File: tests/swagger20_validator/format_validator_test.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        'x-nullable': False,
+        'type': 'string',
+        'format': 'url',
+    }
+    args = (None, prop_schema['format'], None, prop_schema)
+    list(format_validator(minimal_swagger_spec, *args))
+    m_format_validator.assert_called_once_with(*args)
+
+
+@pytest.mark.parametrize(
```
Comment (adamhadani) [inline tests/swagger20_validator/format_validator_test.py]: ping @macisamuele added this per your reference lmk if looks good!

**Comment by adamhadani** [comment]
ping @macisamuele - addressed both your points. I do notice currently two failures on Travis which seem unrelated to PR changes: the coverage test task and the py26 (failing on import of `functools32`)

**Comment by macisamuele** [comment]
@adamhadani Thanks a lot for your contribution.
I have double checked your code change and still I'm not able to understand how your code modifies the initial base behaviour.
I have executed the new test ``test_validate_object_with_different_format_configurations`` against ``origin/master`` and it passes. I think that the test is not covering your code or if the test matches your use-case the code change could be not needed. Could you check double check it?


```
(py35) YELP-MACI:bravado-core maci$ git diff origin/master
diff --git a/tests/swagger20_validator/format_validator_test.py b/tests/swagger20_validator/format_validator_test.py
new file mode 100644
index 0000000..23ea491
--- /dev/null
+++ b/tests/swagger20_validator/format_validator_test.py
@@ -0,0 +1,39 @@
+import pytest
+from jsonschema import ValidationError
+from mock import patch
+from bravado_core.validate import validate_object
+
+
+@pytest.mark.parametrize(
+    'value, format_, x_nullable, expect_exception',
+    (
+        [{'prop': 'a813d449-21ff-4348-8262-c9c067151eb2'}, 'uuid', False, False],
+        [{'prop': None}, 'uuid', False, True],
+        [{'prop': None}, 'uuid', True, False],
+    )
+)
+def test_validate_object_with_different_format_configurations(
+        minimal_swagger_spec, value, format_, x_nullable, expect_exception):
+    minimal_swagger_spec.spec_dict['definitions']['obj'] = {
+        'properties': {
+            'prop': {
+                'type': 'string',
+                'format': format_,
+                'x-nullable': x_nullable
+            }
+        }
+    }
+    captured_exception = None
+    try:
+        validate_object(
+            swagger_spec=minimal_swagger_spec,
+            object_spec=minimal_swagger_spec.spec_dict['definitions']['obj'],
+            value=value,
+        )
+    except ValidationError as e:
+        captured_exception = e
+
+    if not expect_exception:
+        assert captured_exception is None
+    else:
+        assert captured_exception.message == '{0} is not of type \'string\''.format(value['prop'])
(py35) YELP-MACI:bravado-core maci$ py.test tests/swagger20_validator/format_validator_test.py
================================= test session starts =================================
platform darwin -- Python 3.5.2, pytest-3.0.6, py-1.4.32, pluggy-0.4.0
rootdir: /Users/maci/pg/bravado-core, inifile:
collected 3 items

tests/swagger20_validator/format_validator_test.py ...

============================== 3 passed in 0.05 seconds ===============================
```

File: CHANGELOG.rst (hunk 1 above)
Comment (macisamuele) [inline CHANGELOG.rst]: 👍

**Comment by macisamuele** [comment]
@adamhadani About python 2.6 failures ... don't worry ... #147 will fix it + we'll remove support for deprecated python soon!

**Comment by adamhadani** [comment]
@macisamuele  thanks for feedback - i'm currently trying to reproduce my original error to figure out if this PR still needed - possibly some other changes in jsonschema/bravado-core along the way fixed this issue but I wanna make sure before we move forward one way or another, will let you know concusions

**Comment by pokey** [comment]
@macisamuele good catch.  Looks like the error only arises when using a custom validator.  I've updated the test to use a custom validator, and confirmed that it fails when run against master:

```
(bravado) Brandons-MacBook-Pro~/src/bravado-core(master↑1|✔) % git diff upstream/master
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
added: tests/swagger20_validator/format_validator_test.py
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
@ format_validator_test.py:4 @
import pytest
from jsonschema import ValidationError

from bravado_core.exception import SwaggerValidationError
from bravado_core.formatter import SwaggerFormat
from bravado_core.validate import validate_object


def validate_dummy(dummy_string):
    if dummy_string != 'dummy':
        raise SwaggerValidationError('dummy')


DummyFormat = SwaggerFormat(
    format="dummy",
    to_wire=lambda x: x,
    to_python=lambda x: x,
    validate=validate_dummy,
    description="dummy format",
)


@pytest.mark.parametrize(
    'value, format_, x_nullable, expect_exception',
    (
        [{'prop': 'dummy'}, 'dummy', False, False],
        [{'prop': 'hello'}, 'dummy', False, True],
        [{'prop': None}, 'dummy', False, True],
        [{'prop': None}, 'dummy', True, False],
    )
)
def test_validate_object_with_different_format_configurations(
        minimal_swagger_spec, value, format_, x_nullable, expect_exception):
    minimal_swagger_spec.spec_dict['definitions']['obj'] = {
        'properties': {
            'prop': {
                'type': 'string',
                'format': format_,
                'x-nullable': x_nullable
            }
        }
    }
    minimal_swagger_spec.register_format(DummyFormat)
    captured_exception = None
    try:
        validate_object(
            swagger_spec=minimal_swagger_spec,
            object_spec=minimal_swagger_spec.spec_dict['definitions']['obj'],
            value=value,
        )
    except ValidationError as e:
        captured_exception = e

    if not expect_exception:
        assert captured_exception is None
    else:
        assert (
            captured_exception.message == '{0} is not a \'dummy\''.format(repr(value['prop'])) or
            captured_exception.message == '{0} is not of type \'string\''.format(repr(value['prop']))
        )
(bravado) Brandons-MacBook-Pro~/src/bravado-core(master↑1|✔) % py.test tests/swagger20_validator/format_validator_test.py
================================================================= test session starts =================================================================
platform darwin -- Python 3.6.0, pytest-3.0.6, py-1.4.32, pluggy-0.4.0
rootdir: /Users/pokey/src/bravado-core, inifile:
collected 4 items

tests/swagger20_validator/format_validator_test.py ...F

====================================================================== FAILURES =======================================================================
_________________________________ test_validate_object_with_different_format_configurations[value3-dummy-True-False] __________________________________

minimal_swagger_spec = <bravado_core.spec.Spec object at 0x10a11fb38>, value = {'prop': None}, format_ = 'dummy', x_nullable = True
expect_exception = False

    @pytest.mark.parametrize(
        'value, format_, x_nullable, expect_exception',
        (
            [{'prop': 'dummy'}, 'dummy', False, False],
            [{'prop': 'hello'}, 'dummy', False, True],
            [{'prop': None}, 'dummy', False, True],
            [{'prop': None}, 'dummy', True, False],
        )
    )
    def test_validate_object_with_different_format_configurations(
            minimal_swagger_spec, value, format_, x_nullable, expect_exception):
        minimal_swagger_spec.spec_dict['definitions']['obj'] = {
            'properties': {
                'prop': {
                    'type': 'string',
                    'format': format_,
                    'x-nullable': x_nullable
                }
            }
        }
        minimal_swagger_spec.register_format(DummyFormat)
        captured_exception = None
        try:
            validate_object(
                swagger_spec=minimal_swagger_spec,
                object_spec=minimal_swagger_spec.spec_dict['definitions']['obj'],
                value=value,
            )
        except ValidationError as e:
            captured_exception = e

        if not expect_exception:
>           assert captured_exception is None
E           assert <ValidationError: "None is not a 'dummy'"> is None

tests/swagger20_validator/format_validator_test.py:55: AssertionError
========================================================= 1 failed, 3 passed in 0.08 seconds ==========================================================
(bravado) Brandons-MacBook-Pro~/src/bravado-core(master↑1|✔) %
```

**Comment by macisamuele** [comment]
@adamhadani , @pokey thanks a lot for your contribution on the project.
I'm going to merge your changes!

**Pull request merged**